Clean/Circuit/Basic.lean

```diff
@@ -594,11 +594,17 @@ attribute [circuit_norm] Vector.map_mk List.map_toArray List.map_cons List.map_n
 attribute [circuit_norm] Vector.append_singleton Vector.mk_append_mk Vector.push_mk
   Array.append_singleton Array.append_empty List.push_toArray
   List.nil_append List.cons_append List.append_toArray
-  Vector.mapFinRange Vector.mapRange Vector.toArray_push Array.push_toList List.append_assoc
+  Vector.mapRange_zero Vector.mapRange_succ Vector.toArray_push Array.push_toList List.append_assoc
   Vector.eq_mk Vector.mk_eq
 
--- simplify `vector.get 0` (which occurs in ProvableType definitions)
--- TODO handle other small indices as well
+-- simplify Vector.mapFinRange
+attribute [circuit_norm] Vector.mapFinRange_succ Vector.mapFinRange_zero
+    Nat.cast_zero Nat.cast_one Nat.cast_ofNat Fin.coe_eq_castSucc Fin.reduceCastSucc
+
+-- simplify stuff like (3 : Fin 8).val = 3 % 8
+attribute [circuit_norm] Fin.coe_ofNat_eq_mod
```

Comment:
unrelated to other stuff in this PR, I played with `circuit_norm` attributes a bit to improve simplification of commonly occurring expressions